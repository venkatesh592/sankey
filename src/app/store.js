import { configureStore } from '@reduxjs/toolkit';
import sankeyReducer from '../features/sankey/sankeySlice';

export default configureStore({
  reducer: {
    options: sankeyReducer,
  },
});
