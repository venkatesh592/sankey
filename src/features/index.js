import { DataLink } from './data-link/DataLink'
import { DataLinkList } from './data-link-list/DataLinkList'
import { DataPoint } from './data-point/DataPoint'
import { DataPointList } from './data-point-list/DataPointList'
import { EditDialog } from './dialog/EditDialog'
import { Header } from './header/Header'
import { Sankey } from './sankey/Sankey'

export {
  DataLink,
  DataLinkList,
  DataPoint,
  DataPointList,
  EditDialog,
  Header,
  Sankey
}
