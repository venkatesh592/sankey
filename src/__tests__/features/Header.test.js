import React from 'react';
import { render, fireEvent, within } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../app/store';
import { useTranslation } from 'react-i18next';
import { Header } from '../../features';

let container = null
beforeEach(() => {
  container = render(
    <Provider store={store}>
      <Header title="Centime" />
    </Provider>
  );
})

test('renders With out fail', () => {
  expect(container.getByText('Centime')).toBeInTheDocument();
})

test('snapshot', () => {
  expect(container.asFragment()).toMatchSnapshot();
})

test('should change language', async () => {
  const { i18n } = useTranslation()
  fireEvent.mouseDown(container.getByRole('button'))
  const listbox = within(container.getByRole('listbox'))
  fireEvent.click(listbox.getByText(/French/i))
  expect(listbox).toBeTruthy()
})

afterAll(() => {
  jest.clearAllMocks()
})
