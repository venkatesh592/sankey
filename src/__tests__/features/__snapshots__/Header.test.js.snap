// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <a
      href="#default"
    >
      Centime
    </a>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline select"
      data-testid="slelect-lang"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-label demo-simple-select"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="demo-simple-select"
        role="button"
        tabindex="0"
      >
        English
      </div>
      <input
        type="hidden"
        value="en"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
